secrets/provider/vault/provider_test.go

```diff
@@ -73,7 +73,7 @@ func (s *providerSuite) TestBackendConfig(c *gc.C) {
 			},
 		},
 	}
-	_, err = p.RestrictedConfig(adminCfg, nil, nil, nil)
+	_, err = p.RestrictedConfig(adminCfg, false, nil, nil, nil)
```

Comment:
should have a test with fordrain=true also

Reply:
Yeah, we don't have existing tests to cover policies. I will refactor the test suite to add tests to cover all policies.